File: [BOJ] 백준 풀이/단계별로 풀어보기 (step by step)/#4 1차원 배열/8958_OX퀴즈 (성공).cpp
#include <iostream>
#include <cstring>

int main()
{
    int N;
    char arr[80];
    int count, score, len;

    std::cin >> N;

    for (int i = 0; i < N; i++)
    {
        score = 0;
        count = 1;

        std::cin >> arr;
        len = strlen(arr);

        for (int j = 0; j < len; j++)
        {
            if (arr[j] == 'O')
            {
                score += count;
                count++;
            }

            else if (arr[j] == 'X')
                count = 1;
        }

        std::cout << score << "\n";
    }
}

File: 백준 풀이/~23.12.31/#04 1차원 배열/8958_OX퀴즈 (성공).py
N = int(input())
for i in range(N):
    arr = list(input())
    sum = 0
    count = 1

    for j in arr:
        if j == 'O':
            sum += count
            count += 1
        else:
            count = 1
            
    print(sum)
